docs/the-new-architecture/use-app-template.md

```diff
@@ -1,7 +1,134 @@
+Application**, and resume following this guide.
+
+**Note**: The New Architecture cannot currently be enabled in an Expo app.
+
+## Creating a new application
+
+<RemoveGlobalCLI />
+
+If you already have your development environment set up, create a new React Native project from the template:
+
+```shell
+npx react-native init AwesomeProject
+```
+
+If specifying a versioned template, use version 0.68 or later.
+
+## Configuration
+
+Follow the steps below to enable the New Architecture and build the app.
+
+### Enable Hermes
+
+Hermes is an open-source JavaScript engine optimized for React Native. [Hermes will be the default engine in the future](https://github.com/reactwg/react-native-new-architecture/discussions/4), and we highly recommend you use it.
+
+Please [follow the instructions on the React Native website](hermes.md) in order to enable Hermes in your application.
+
+### Enable the New Architecture
+
+#### Target OS
+
+<Tabs groupId="platform" defaultValue={constants.defaultPlatform} values={constants.platforms} className="pill-tabs">
+<TabItem value="ios">
+
+Navigate to the `ios` directory and run the following:
+
+```shell
+# from `ios` directory
+RCT_NEW_ARCH_ENABLED=1 bundle exec pod install
+```
+
+Then build and run the app as usual:
+
+```shell
+npx react-native run-ios
+```
+
+> **Tip**: You need to re-run `pod install` each time a dependency with native
+> code changes. Make this command easier to run by adding it to `scripts` to your
+> project's `package.json` file:
+>
+> ```
+> "scripts": {
+>   "pod-install": "RCT_NEW_ARCH_ENABLED=1 bundle exec pod install"
+> }
+> ```
+>
+> and run it with `yarn pod-install`.
+
+#### Troubleshooting
+
+##### `react-native run-ios` fails
+
+If you see a build failure from `react-native run-ios`, there may be cached files from a previous build with
+the old architecture. Clean the build cache and try again:
+
+1. Open the project `ios/project.xcworkspace` in Xcode
+2. In XCode, choose Product > Clean Build Folder
+3. Try building and running the app again
+
+</TabItem>
+<TabItem value="android">
+
+Set the `newArchEnabled` property to `true` by **either**:
+
+- Changing the corresponding line in `android/gradle.properties`
+- Setting the environment variable `ORG_GRADLE_PROJECT_newArchEnabled=true`
+
+Then build and run the app as usual:
+
+```shell
+npx react-native run-android
+```
+
+> **Note**: You may notice longer build times with the New Architecture, due to
+> additional step of C++ compilation with the Android NDK. To improve your build
+> time, see
+> [Speeding Up Your Build Phase](docs/build-speed.md).
+
+</TabItem>
+</Tabs>
+
+### Confirming the New Architecture is in use
+
+After you build and run the app, when Metro serves the JavaScript bundle, you
+should see `"fabric": true` in the Metro logs:
+
+<img src="/docs/assets/metro-new-arch.png" alt="Metro shows fabric: true" width="600" />
+
+### Want to know more?
```

Comment:
I think that the answer is yes. Of course, they have to do the opposite steps that we are suggesting here:
* for iOS: run `RCT_NEW_ARCH_ENABLED=0 pod install` (or just `pod install`, given that the default value for the flag is `0`).
* for Android: switch the `newArchEnabled` and the `ORG_GRADLE_PROJECT_newArchEnabled` to `false` and relaunch the app.

From the Template perspective, this should be enough. The template works with both the old and the New Architecture out-of-the-box. If the devs started adding dependencies that only supports the New Architecture, their app won't work, clearly.

Reply:
If it is just switching those flags, I think we can probably leave it alone for now unless anyone's confused by it, but let me know if you think it would be good to add!

Reply:
Let's omit that, though. I'd rather have a leaner documentation and it looks reasonable that the users try to turn off those switches to go back to the Old Architecture, even without an explicit paragraph!